### DISCUSSION TITLE
Can't build a sample app due to cocoas pods error.

### DISCUSSION THREAD

**Discussion by gonnashred** [flyerhq/flutter_chat_ui#110] [Q&A] [answered]
Hi All,

Frairly new to flutter, and trying to get a sample up and running using flutter_chat_ui and getting an error in cocoapods.

Environment is:
Brand new Mac mini M1 
Visual studio with the Flutter development extensions installed
Ruby 2.6.3p62

When trying to build a trivial sample, I get a cocoapods error:

Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
lib/main.dart:1
CocoaPods' output:
↳
      Preparing
    Analyzing dependencies
    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)
    Fetching external sources
    -> Fetching podspec for `Flutter` from `Flutter`
    -> Fetching podspec for `url_launcher` from `.symlinks/plugins/url_launcher/ios`
    Resolving dependencies of `Podfile`
Error output from CocoaPods:
↳
    /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': dlopen(/Library/Ruby/Gems/2.6.0/gems/ffi-1.15.4/lib/ffi_c.bundle, 9): no suitable image found.  Did find: (LoadError)
    	/Library/Ruby/Gems/2.6.0/gems/ffi-1.15.4/lib/ffi_c.bundle: mach-o, but wrong architecture
    	/Library/Ruby/Gems/2.6.0/gems/ffi-1.15.4/lib/ffi_c.bundle: mach-o, but wrong architecture - /Library/Ruby/Gems/2.6.0/gems/ffi-1.15.4/lib/ffi_c.bundle
    	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
    	from /Library/Ruby/Gems/2.6.0/gems/ffi-1.15.4/lib/ffi.rb:5:in `rescue in <top (required)>'
    	from /Library/Ruby/Gems/2.6.0/gems/ffi-1.15.4/lib/ffi.rb:2:in `<top (required)>'

Thanks in advance for any help you can offer. Great effort you guys have going here.

Thanks,
Adam

**Comment by demchenkoalex** [comment, marked as answer]
Hey @gonnashred I have M1 myself and the error tells you that `ffi` Ruby dependency has no support for ARM architecture, you can also easily google it, but basically on M1 you still can't successfully perform `pod install` without running in Rosetta mode.

Find your terminal app, right click on it, `Get Info`, and check `Open using Rosetta`, then you can navigate to the `example/ios` folder and run `pod install` (from the terminal in `Rosetta` mode, that console in VSCode will be in the ARM mode, unless you run all VSCode in Rosetta). After successful `pod install` you can normally run the app from VSCode (ARM). This is my workflow.

There are alternative solutions, again can be easily found in Google (keep in mind that alternative solutions still will be somewhat connected with running in Rosetta, so it is the easiest thing, described above)

**Reply by gonnashred** [reply]
This got me most of the way there. Before it would fully work, I had to run `sudo gem install ffi` to get the correct gem on. Then your above answer worked perfect. Thank you.
